File: apps/user-app-web/src/lib/api/paymentApi.ts
import { apiClient } from './client';
import { unwrapResponse, type BffResponse } from './bffParser';

export interface PreparePaymentRequest {
  amount: number;
  orderName: string;
  bookingId?: number;
}

export interface PreparePaymentResponse {
  orderId: string;
  amount: number;
  orderName: string;
}

export interface ConfirmPaymentRequest {
  paymentKey: string;
  orderId: string;
  amount: number;
}

export interface ConfirmPaymentResponse {
  paymentId: number;
  orderId: string;
  paymentKey: string;
  amount: number;
  status: string;
}

export interface PaymentStatusResponse {
  id: number;
  orderId: string;
  paymentKey: string | null;
  amount: number;
  status: string;
  bookingId: number | null;
}

export interface ConfirmSplitPaymentRequest {
  paymentKey: string;
  orderId: string;
  amount: number;
}

export const paymentApi = {
  preparePayment: async (data: PreparePaymentRequest): Promise<PreparePaymentResponse> => {
    const response = await apiClient.post<BffResponse<PreparePaymentResponse>>(
      '/api/user/payments/prepare',
      data,
    );
    return unwrapResponse(response.data);
  },

  confirmPayment: async (data: ConfirmPaymentRequest): Promise<ConfirmPaymentResponse> => {
    const response = await apiClient.post<BffResponse<ConfirmPaymentResponse>>(
      '/api/user/payments/confirm',
      data,
    );
    return unwrapResponse(response.data);
  },

  confirmSplitPayment: async (data: ConfirmSplitPaymentRequest): Promise<ConfirmPaymentResponse> => {
    const response = await apiClient.post<BffResponse<ConfirmPaymentResponse>>(
      '/api/user/payments/split/confirm',
      data,
    );
    return unwrapResponse(response.data);
  },

  getPaymentByOrderId: async (orderId: string): Promise<PaymentStatusResponse> => {
    const response = await apiClient.get<BffResponse<PaymentStatusResponse>>(
      `/api/user/payments/order/${orderId}`,
    );
    return unwrapResponse(response.data);
  },
};

File: apps/user-app-android/app/src/main/java/com/parkgolf/app/data/remote/api/PaymentApi.kt
package com.parkgolf.app.data.remote.api

import com.parkgolf.app.data.remote.dto.common.ApiResponse
import com.parkgolf.app.data.remote.dto.payment.AbandonPaymentRequest
import com.parkgolf.app.data.remote.dto.payment.ConfirmPaymentRequest
import com.parkgolf.app.data.remote.dto.payment.ConfirmPaymentResponse
import com.parkgolf.app.data.remote.dto.payment.ConfirmSplitPaymentResponse
import com.parkgolf.app.data.remote.dto.payment.PaymentStatusResponse
import com.parkgolf.app.data.remote.dto.payment.PreparePaymentRequest
import com.parkgolf.app.data.remote.dto.payment.PreparePaymentResponse
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface PaymentApi {

    @POST("api/user/payments/prepare")
    suspend fun preparePayment(@Body request: PreparePaymentRequest): ApiResponse<PreparePaymentResponse>

    @POST("api/user/payments/confirm")
    suspend fun confirmPayment(@Body request: ConfirmPaymentRequest): ApiResponse<ConfirmPaymentResponse>

    @POST("api/user/payments/split/confirm")
    suspend fun confirmSplitPayment(@Body request: ConfirmPaymentRequest): ApiResponse<ConfirmSplitPaymentResponse>

    @GET("api/user/payments/order/{orderId}")
    suspend fun getPaymentByOrderId(@Path("orderId") orderId: String): ApiResponse<PaymentStatusResponse>

    /**
     * 결제 중단 통지 (Toss 결제 실패/취소)
     * payment.status=ABORTED + outbox booking.paymentFailed → PAYMENT_FAILED Saga
     */
    @POST("api/user/payments/{orderId}/abandon")
    suspend fun abandonPayment(
        @Path("orderId") orderId: String,
        @Body request: AbandonPaymentRequest,
    ): ApiResponse<PaymentStatusResponse>
}
